client/asset/btc/btc.go

```diff
@@ -610,17 +610,25 @@ func (btc *ExchangeWallet) SignMessage(coin asset.Coin, msg dex.Bytes) (pubkeys,
 	if err != nil {
 		return nil, nil, err
 	}
+	if len(tx.Details) == 0 {
+		return nil, nil, fmt.Errorf("no tx details found for %s = %s:%v", coin, output.txHash.String(), output.vout)
+	}
 	for _, txDetails := range tx.Details {
 		if txDetails.Vout == output.vout &&
```

Comment:
That's a good question, and it sent me down a little rabbit hole to find that this will not work in it's current form for mutli-sig. But I do see the path forward. Mind if I follow up?

Reply:
Sure.  Although I'm still open to disallowing multisig (except for p2sh multisig).

Reply:
That's my intention. The `gettransaction` will return info for P2SH multi-sig addresses that have been added with `addmultisigaddress`. Unfortunately, P2SH address are not compatible with `privkeyforaddress`, but I can get its pubkeys with `getaddressinfo`, so I think I should be able to make it work.